src/inference/src/dev/core_impl.cpp

```diff
@@ -272,21 +263,22 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
     auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<ov::ICompiledModel> res;
     auto cacheManager =
-        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
-            ._cacheManager;
+        coreConfig.get_cache_config_for_device(plugin, ov::cache_dir.name(), parsed._config)._cacheManager;
```

Comment:
It can be somethig like this:
```
// Merge with user settings and get config
auto context = config_manager->merge(plugin, config_with_batch);
auto cacheManager = context.get_property(ov::cache_manager).as<std::shared_ptr<CacheManager>>();
auto device = context.get_property(ov::device::name).as<std::string>();
...
```